kie-server-parent/kie-server-api/pom.xml

```diff
@@ -24,11 +24,18 @@
       <artifactId>kie-api</artifactId>
       <!-- scope>provided</scope -->
     </dependency>
+
+    <!-- we need to find a way to remove the following 2 dependencies from this jar -->
     <dependency>
       <groupId>org.drools</groupId>
       <artifactId>drools-core</artifactId>
       <!-- scope>provided</scope -->
     </dependency>
+    <dependency>
+      <groupId>org.optaplanner</groupId>
+      <artifactId>optaplanner-core</artifactId>
+      <!-- scope>provided</scope -->
```

Comment:
As I mentioned before, I would rather try to remove the dependency completely, but we can try with optional for now.